### PR TITLE
Diagonal gate and arthas heart attack

### PR DESCRIPTION
**Pull request by SebastianTanoni**

### CODE REVIEW

File: src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs [hunk 1]
```diff
@@ -21,28 +21,22 @@ public QuestDestroyStratholme(Capital stratholme, LegendaryHero arthas) : base("
...
-      } while (!IsPointValidForArthas(randomPoint));
+      var arthas = _arthas.Unit;
+
+      if (arthas == null || !UnitAlive(arthas))
+        return;
 
-      if (_arthas.Unit == null || !UnitAlive(_arthas.Unit))
-        _arthas.ForceCreate(LordaeronSetup.Lordaeron?.Player ?? Player(PLAYER_NEUTRAL_AGGRESSIVE), randomPoint, 270);
-      
-      completingFaction.AddPower(new PingPower(_arthas, "Eye of the Lich King", 5, 5));
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs:45]: This is the only instance of `PingPower` so you should delete that.

**Review by YakaryBovine — CHANGES_REQUESTED** [review]
Everything seems good, just need to delete the now unused Power.

File: src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs [hunk 2]
```diff
@@ -21,28 +21,22 @@ public QuestDestroyStratholme(Capital stratholme, LegendaryHero arthas) : base("
...
-    private static bool IsPointValidForArthas(Point whichPoint) =>
-      !whichPoint.IsPathable(PATHING_TYPE_FLOATABILITY) && whichPoint.IsPathable(PATHING_TYPE_WALKABILITY);
-
     /// <inheritdoc />
-    protected override string RewardFlavour => "Prince Arthas could not protect the people of Stratholme. The Lich King's hold over him grows stronger.";
+    protected override string RewardFlavour => "Having failed to protect his people, Arthas seizes the cursed runeblade Frostmourne as the instrument of his vengeance. The malevolence of the blade overwhelms him. Arthas is now a loyal Death Knight of the Scourge, and will soon become its greatest champion.";
 
     /// <inheritdoc />
-    protected override string RewardDescription => "Gain the power Eye of the Lich King, which allows you to identify where Arthas is at any time";
+    protected override string RewardDescription => $"Learn to train {_arthas.Name} from the {GetObjectName(Constants.UNIT_UAOD_ALTAR_OF_DARKNESS)}";
```
Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs:28]: Should indicate that it causes Arthas to die permanently.

Comment (YakaryBovine) [inline src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs:28]: The quest doesn't actually give you Arthas... Have you tested this?
image

Comment (SebastianTanoni) [reply src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs:28]: i tested if it perma killed arthas, i didnt check if scourge gained him. thats my bad

Comment (YakaryBovine) [reply src/WarcraftLegacies.Source/Quests/Scourge/QuestDestroyStratholme.cs:28]: Sweet!

**Review by YakaryBovine — APPROVED** [review]
LGTM

**Review by YakaryBovine — APPROVED** [review]

**Pull request merged**